coordinape/coordinape
# Defunct - Proposal for local dev environment

Requires: Hasura cli version >= 2.1.1

This is a proposed setup for Hasura local development w/ all sub systems

### Goals:
 - Laravel manages the DB migrations
   - Easily create larvel migrations and develop in Hasura concurrently
 - `setup.sh` + `docker-compose up` + `yarn run seed-db 0x00ME00` + `vercel dev`
   -  **Run everything needed** for all our subsystems to run **locally**, including auth connecting & laravel
 - Enable integration testing in CI
 - Enable contributors to easily work on big circles (100+)

### Notice
 - Using hasura database name "default" is preferable
   - w/ env variable `PG_DATABASE_URL` this is the default initialization
   - It will be consistent and semantically correct naming in Prod & Staging & Local 

### TODO:
 - Merge latest Hasura metadata
 - Make sure all the services talk to each other correctly, and update .env.example
 - Rewrite documentation for dev setup
   - Include documentation for developing with staging only
 - Create a vercel function (gqty or prisma?) to initialize circles in a parametrizable way
   - Perhaps available in prod/staging w/ super admin credentials 
 - Make sure it works for other devs (non M1 setups)
 
 
 ### Out of scope but seems doable
 - AWS S3 emulation w/ docker compose
 - hardhat in docker compose
   - (containerize mainnet fork??)
